### PR TITLE
Fix test body motion

### PR DESCRIPTION
**Pull request by Ughuuu**
- Fixes https://github.com/fabriceci/godot-rapier2d/issues/16
- Fixes https://github.com/fabriceci/godot-rapier2d/issues/7

### CODE REVIEW

File: src/rapier_body_utils_2d.cpp [hunk 1]
```diff
@@ -99,7 +90,9 @@ bool RapierBodyUtils2D::body_motion_recover(
 				Vector2 b(contact.point2.x, contact.point2.y);
 
 				// Compute plane on b towards a.
-				Vector2 n = (a - b).normalized();
+				Vector2 n = Vector2(contact.normal1.x, contact.normal1.y);
```
Comment (fabriceci) [inline src/rapier_body_utils_2d.cpp:93]: I find a-b clearer, are you sure that it's normal1 that should be used and not 2?

File: src/rapier_body_utils_2d.cpp [hunk 2]
```diff
@@ -99,7 +90,9 @@ bool RapierBodyUtils2D::body_motion_recover(
...
+				// Move it outside as to fit the margin
```
Comment (fabriceci) [inline src/rapier_body_utils_2d.cpp:94]: I'm not comfortable with the idea of changing the algorithm, why did you do that?

File: src/rapier_body_utils_2d.cpp [hunk 3]
```diff
@@ -15,7 +14,7 @@ bool RapierBodyUtils2D::body_motion_recover(
 		Rect2 &p_body_aabb) {
```
Comment (fabriceci) [inline src/rapier_body_utils_2d.cpp:14]: If you want to pass the marign_aabb (which I don't think you can, see the comment below), rename this to p_margin_aabb

File: src/rapier_body_utils_2d.cpp [hunk 4]
```diff
@@ -110,12 +103,10 @@ bool RapierBodyUtils2D::body_motion_recover(
 				}
 			}
 		}
-
 		if (recovered) {
 			p_recover_motion += recover_step;
 			p_transform.columns[2] += recover_step;
 			p_body_aabb.position += recover_step;
```
Comment (fabriceci) [inline src/rapier_body_utils_2d.cpp:109]: Again, I'm just looking at the changes, pretty hard to do, but the principle is to move the body, except here you're moving the margin.

see at the end (the body position here is not moved by the recover):

```
		// Get the rest info in unsafe advance
		Vector2 unsafe_motion = p_motion * best_unsafe;
		body_transform.columns[2] += unsafe_motion;
		body_aabb.position += unsafe_motion;
```

File: src/rapier_body_utils_2d.cpp [hunk 5]
```diff
@@ -38,19 +37,11 @@ bool RapierBodyUtils2D::body_motion_recover(
 		body_shape_handle = p_body.get_shape(0)->get_rapier_shape();
 	}*/
 
-	real_t margin = MAX(p_margin, TEST_MOTION_MARGIN_MIN_VALUE);
-	real_t min_contact_depth = margin * TEST_MOTION_MIN_CONTACT_DEPTH_FACTOR;
-
-	// Undo the currently transform the physics server is aware of and apply the provided one
-	p_body_aabb = p_transform.xform(p_body_aabb);
```
[the remark this answers is not in the document]
Comment (Ughuuu) [reply src/rapier_body_utils_2d.cpp:45]: Moving it outside of the function so it's more obvious that it's a side effect. Moved it outside of the steps as it always happens and it's later used in step 3

File: src/rapier_body_utils_2d.cpp (hunk 1 above)
Comment (Ughuuu) [reply src/rapier_body_utils_2d.cpp:93]: A-b is cleaner, but normal is already computed, so its more efficient like this. Y, its normal 1.

File: src/rapier_body_utils_2d.cpp (hunk 2 above)
Comment (Ughuuu) [reply src/rapier_body_utils_2d.cpp:94]: Is it not the same? Normal1 and a-b normalized is the same value, and its already computed by the function. So i just used that to save computation. Normalized does sqrt inside, so its expensive.

File: src/rapier_body_utils_2d.cpp [hunk 6]
```diff
@@ -38,19 +37,11 @@ bool RapierBodyUtils2D::body_motion_recover(
 		body_shape_handle = p_body.get_shape(0)->get_rapier_shape();
 	}*/
 
-	real_t margin = MAX(p_margin, TEST_MOTION_MARGIN_MIN_VALUE);
```
[the remark this answers is not in the document]
Comment (Ughuuu) [reply src/rapier_body_utils_2d.cpp:41]: Fair, will readd.

[the remark this answers is not in the document]
Comment (Ughuuu) [reply src/rapier_body_utils_2d.cpp:41]: Ah, no, actually that didn't get deleted, moved it a level higher, in test_body_motion before the steps as it's more of a validation step. If it's done in step 1, it hides the intention.

**Pull request merged**